pkg/auth/oauth/external/keycloak/keycloak.go

```diff
@@ -0,0 +1,132 @@
+package keycloak
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/RangelReale/osincli"
+	"github.com/golang/glog"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/auth/oauth/external"
+)
+
+type provider struct {
+	Realm          string `json:"realm"`
+	RealmPublicKey string `json:"realm-public-key"`
+	ClientID       string `json:"resource"`
+	Credentials    struct {
+		Secret string `json:"secret"`
+	} `json:"credentials"`
+	AuthServerURL string `json:"auth-server-url"`
```

Comment:
Good point.  The documentation doesn't cover any of this.  I suspect this is left up to the user to ensure they've configured their key/trust stores appropriately.  I'll walk through their ssl example and see what shakes out.  I should probably also test ssl on the OS side of things as well.